sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -139,6 +138,54 @@ class DetermineTableStats(session: SparkSession) extends Rule[LogicalPlan] {
   }
 }
 
+case class DeterminePartitionedTableStats(
```

Comment:
is it kind of a `PruneFileSourcePartitions` for hive tables? shall we call it `PruneHiveTablePartitions`?